components/core/src/crypto/artifact.rs

```diff
@@ -1,33 +1,49 @@
-use super::{hash,
-            keys::parse_name_with_rev,
-            SigKeyPair,
-            HART_FORMAT_VERSION,
-            SIG_HASH_TYPE};
-use crate::error::{Error,
-                   Result};
-use sodiumoxide::crypto::sign;
+use crate::{crypto::{keys::{Key,
+                            KeyCache,
+                            NamedRevision,
+                            SecretOriginSigningKey},
+                     Blake2bHash,
+                     HART_FORMAT_VERSION,
+                     SIG_HASH_TYPE},
+            error::{Error,
+                    Result}};
 use std::{fs::File,
           io::{self,
                prelude::*,
+               BufRead,
                BufReader,
                BufWriter},
           path::Path};
 
+pub struct ArtifactHeader {
+    format:    String,
+    signer:    NamedRevision,
+    hash_type: String,
+    signature: Vec<u8>,
+}
+
+impl ArtifactHeader {
+    pub fn format(&self) -> &String { &self.format }
+
+    pub fn signer(&self) -> &NamedRevision { &self.signer }
+
+    pub fn hash_type(&self) -> &String { &self.hash_type }
+
+    pub fn signature_raw(&self) -> String { base64::encode(&self.signature) }
```

Comment:
What makes this signature 'raw'?

Reply:
:thinking: This was naming that existed in the original code. It seems to refer to the signature as it is pulled directly out of the HART header. So it is "raw" in that sense, to be further processed by base64 decoding it. However, I tend to think of the actual bytes of the signature as what is "raw", with base64 _encoding_ being the processing applied to it to reach the form we see in the header.

Thank you for catching this. I'll see about renaming to something a bit more straightforward (perhaps something like `encoded_signature`.)

Reply:
Addressed in https://github.com/habitat-sh/habitat/commit/09e76a58b042afb35e2303265adce92babeb62be